.travis.yml

```diff
@@ -1,45 +1,59 @@
+install: true
 
-        chmod 600 "$SSH_KEY_FILE" && printf "%s\n" \
-              "Host github.com" \
-              "  IdentityFile $SSH_KEY_FILE" \
-              "  LogLevel ERROR" >> ~/.ssh/config;
+script:
+- test -f "$DOCKER_DIR/image.tar" && docker load -i "$DOCKER_DIR/image.tar" || true
+- docker build -t core:base --target base .
+- docker build -t core:dist --target dist --build-arg VCS_BRANCH="$TRAVIS_BRANCH" --build-arg VCS_COMMIT="$TRAVIS_COMMIT" .
+- docker build -t core:testing --target testing .
+- docker save -o "$DOCKER_DIR/image.tar" $(docker history -q core:base | grep -v '<missing>') $(docker history -q core:dist | grep -v '<missing>')
+- ./tests/bin/docker-tests.sh --image core:testing
 
-        git config --global user.email "travis@travis-ci.org";
-        git config --global user.name "Travis CI";
-        git config --global push.default simple;
-    fi
-  - if [ "$TRAVIS_BRANCH" == "master" -o  "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
-        bash <(curl -s https://codecov.io/bash) -cF python;
-    fi
-  - if [ "$TRAVIS_TAG" ]; then
-        ./docker/build-trigger.sh Tag "$TRAVIS_TAG" "$BUILD_TRIGGER_URL";
-        ./bin/push-docs.sh "$GIT_REMOTE" tags "$TRAVIS_TAG" "Travis Core Docs Build - ${TRAVIS_BUILD_NUMBER}";
-    fi
-  - if [ "$TRAVIS_EVENT_TYPE" == "push" -a "$TRAVIS_BRANCH" == "master" ]; then
-        ./docker/build-trigger.sh Branch "$TRAVIS_BRANCH" "$BUILD_TRIGGER_URL";
-    fi
-  - if [ "$TRAVIS_EVENT_TYPE" == "push" -a -z "$TRAVIS_TAG" ]; then
-        ./bin/push-docs.sh "$GIT_REMOTE" branches "$TRAVIS_BRANCH" "Travis Core Docs Build - ${TRAVIS_BUILD_NUMBER}";
-    fi
-  
+after_success:
+- |
+  if [ "$TRAVIS_EVENT_TYPE" == "push" -o "$TRAVIS_TAG" ]; then
+      SSH_KEY_FILE=$(mktemp -p $HOME/.ssh/);
+      openssl aes-256-cbc -K $encrypted_55750ae1fbc7_key -iv $encrypted_55750ae1fbc7_iv -in .github_deploy_key.enc -out $SSH_KEY_FILE -d;
+      chmod 600 $SSH_KEY_FILE;
+      printf "%s\n" \
+        "Host github.com" \
+        "  IdentityFile $SSH_KEY_FILE" \
+        "  LogLevel ERROR" >> ~/.ssh/config;
+      git config --global user.email "travis@travis-ci.org";
+      git config --global user.name "Travis CI";
+      git config --global push.default simple;
+  fi
+- if [ "$TRAVIS_BRANCH" == "master" -o  "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
+      bash <(curl -s https://codecov.io/bash) -cF python;
+  fi
+- if [ "$TRAVIS_TAG" ]; then
+      docker tag core:dist scitran/core:$TRAVIS_TAG;
+      docker push scitran/core:$TRAVIS_TAG;
+      ./bin/push-docs.sh "$GIT_REMOTE" tags "$TRAVIS_TAG" "Travis Core Docs Build - $TRAVIS_BUILD_NUMBER";
+  elif [ "$TRAVIS_EVENT_TYPE" == "push" ]; then
+      ./bin/push-docs.sh "$GIT_REMOTE" branches "$TRAVIS_BRANCH" "Travis Core Docs Build - $TRAVIS_BUILD_NUMBER";
+  fi
+- if [ "$TRAVIS_EVENT_TYPE" == "push" -a "$TRAVIS_BRANCH" == "master" ]; then
+      docker tag core:dist scitran/core:latest;
+      docker push scitran/core:latest;
+  fi
```

Comment:
Is there a good way to further clean up the conditionals in the `after_success` block? It's currently rather hard to read, but I do realize there are order dependencies that add complexity. At minimum, let's have consistency in the test order of `$TRAVIS_BRANCH` and `$TRAVIS_EVENT_TYPE`.